core/app/models/spree/reimbursement.rb

```diff
@@ -111,11 +111,14 @@ def perform!(created_by: nil)
 
       if unpaid_amount_within_tolerance?
         reimbursed!
-        reimbursement_success_hooks.each { |h| h.call self }
-        send_reimbursement_email
+        reimbursement_success_hooks.each { |h| h.call self } # TODO: these should become event subscriptions
       else
         errored!
-        reimbursement_failure_hooks.each { |h| h.call self }
+        reimbursement_failure_hooks.each { |h| h.call self } # TODO: these should become event subscriptions
+      end
+      Spree::Event.fire 'reimbursement_perform', reimbursement: self
```

Comment:
@kennyadsl I think an event may suffice, but it depends on what we want to prioritize.

For example, in this case, the mailer processor acts only when the reimbursement processing was successful by using some simple logic: `if reimbursement.reimbursed?`

The rule of thumb I am suggesting to use here is to add events only when strictly necessary, otherwise there may be an event proliferation to the extreme that, for each `if` or `case` statement in a code block, a different event is fired. 
We're talking about two events instead of one here, so there's no real risk of overdoing it, but I think that having a conversation about _when_ adding new events right now may be beneficial. 

Of course limiting events has the downside that the subscriber has to implement some more logic, just like in the example above.

As both approaches have pros and cons, I think there's no clear winner, at least in my mind. What do you think?

Reply:
I'm proposing two events mainly because I saw that in this specific case there are two hooks, one for success and one for failure. If we are suggesting users to replace hooks with events subscription we should at least provide the same hook points that we used to provide before. I mean that right now it's not used into Solidus but it could be something that stores or extensions are using and this is hard to know. 

Generally speaking, I think it's very hard to provide the right balance between too few and too many events.

Reply:
@kennyadsl 👍 
I'll update the code to include two events, I agree in this specific situation it makes more sense having two separate events.